### PR TITLE
Adding MedSAM2 to the model zoo

### PR DESCRIPTION
**Pull request by prernadh**
## What changes are proposed in this pull request?

Adding MedSAM2 to the model ZOO by adding documentation to [this PR](https://github.com/voxel51/fiftyone/pull/4733) by Evatt Harvey Salinger.

## How is this patch tested? If it is not, please explain why.

Manually tested

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

This PR adds MedSAM2 to the FiftyOne Model Zoo. Detailed description in [this PR](https://github.com/voxel51/fiftyone/pull/4733).

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [x] Other: FiftyOne Zoo

## Summary by CodeRabbit

- **New Features**
	- Introduced support for handling the "med-sam" dataset, including specific dataset loading and processing logic.
	- Enhanced model application capabilities for the "med-sam" dataset with tailored conditions.
	- Integrated session launch for the FiftyOne app post-dataset processing.
- **Refactor**
	- Reorganized import statements for improved structure in the utility scripts.

### CODE REVIEW

**Comment by prernadh** [comment]
I will rebase develop before merging :)

File: docs/scripts/make_model_zoo_docs.py
Comment (brimoor) [inline docs/scripts/make_model_zoo_docs.py:167]: Suggestion: move this `elif` block above line 156 so that you can remove `and 'med-SAM' not in tags` from there?

**Review by brimoor — APPROVED** [review]
LGTM from a static review standpoint, just left one suggestion.

Defer to @danielgural for a functional review (I didn't try running the model)

**Pull request merged**
